src/installer/_compat/typing.py

```diff
@@ -0,0 +1,6 @@
+__all__ = ["TYPE_CHECKING"]
+
+try:  # pragma: no cover
+    from typing import TYPE_CHECKING
+except ImportError:  # pragma: no cover
+    TYPE_CHECKING = False
```

Comment:
```suggestion
if False:
    from typing import TYPE_CHECKING
else:
    TYPE_CHECKING = False
```

nit: let's not import typing even on Python 2 (to avoid uninstalling-in-use-module issues if pip vendors this)?